doc/nvim-treesitter.txt

```diff
@@ -449,228 +449,234 @@ Note: This is highly experimental, and folding can break on some types of
 HIGHLIGHTS                                        *nvim-treesitter-highlights*
 
-							     *hl-TSAnnotation*
-`TSAnnotation`
-For C++/Dart attributes, annotations that can be attached to the code to
-denote some kind of meta information.
+The following is a list of highlights groups, the syntactic elements they
+apply to, and some examples.
 
 							      *hl-TSAttribute*
 `TSAttribute`
-(unstable) TODO: docs
+Annotations that can be attached to the code to denote some kind of meta
+information. e.g. C++/Dart attributes.
 
 								*hl-TSBoolean*
 `TSBoolean`
-For booleans.
+Boolean literals: `True` and `False` in Python.
 
 							      *hl-TSCharacter*
 `TSCharacter`
-For characters.
+Character literals: `'a'` in C.
 
 								*hl-TSComment*
 `TSComment`
-For comment blocks.
+Line comments and block comments.
 
 							    *hl-TSConditional*
 `TSConditional`
-For keywords related to conditionnals.
+Keywords related to conditionals: `if`, `when`, `cond`, etc.
 
 							       *hl-TSConstant*
 `TSConstant`
-For constants
+Constants identifiers. These might not be semantically constant.
+E.g. uppercase variables in Python.
 
 							   *hl-TSConstBuiltin*
 `TSConstBuiltin`
-For constant that are built in the language: `nil` in Lua.
+Built-in constant values: `nil` in Lua.
 
 							     *hl-TSConstMacro*
 `TSConstMacro`
-For constants that are defined by macros: `NULL` in C.
+Constants defined by macros: `NULL` in C.
 
 							    *hl-TSConstructor*
 `TSConstructor`
-For constructor calls and definitions: `{}` in Lua, and Java constructors.
+Constructor calls and definitions: `{}` in Lua, and Java constructors.
 
 								  *hl-TSError*
 `TSError`
-For syntax/parser errors.
+Syntax/parser errors. This might highlight large sections of code while the
+user is typing still incomplete code, use a sensible highlight.
 
 							      *hl-TSException*
 `TSException`
-For exception related keywords.
+Exception related keywords: `try`, `except`, `finally` in Python.
 
 								  *hl-TSField*
 `TSField`
-For fields.
+Object and struct fields.
 
 								  *hl-TSFloat*
 `TSFloat`
-For floats.
+Floating-point number literals.
 
 							       *hl-TSFunction*
 `TSFunction`
-For function (calls and definitions).
+Function calls and definitions.
 
 							    *hl-TSFuncBuiltin*
 `TSFuncBuiltin`
-For builtin functions: `table.insert` in Lua.
+Built-in functions: `print` in Lua.
 
 							      *hl-TSFuncMacro*
 `TSFuncMacro`
-For macro defined fuctions (calls and definitions): each `macro_rules` in
+Macro defined functions (calls and definitions): each `macro_rules` in
 Rust.
 
 								*hl-TSInclude*
 `TSInclude`
-For includes: `#include` in C, `use` or `extern crate` in Rust, or `require`
-in Lua.
+File or module inclusion keywords: `#include` in C, `use` or `extern crate` in
+Rust.
 
 								*hl-TSKeyword*
 `TSKeyword`
-For keywords that don't fall in previous categories.
+Keywords that don't fit into other categories.
 
```

Comment:
Maybe this section could be moved to after the explanation of the other keyword hl groups, so it makes more sense about what the other categories this refers to.

Reply:
The list is sorted alphabetically like `:h highlight-default`.

If highlight groups are still being added I think it's best to keep it that way for now. Afterwards it might make sense to sort them differently (like `:h group-name`).